rmf_traffic_editor/gui/editor.cpp

```diff
@@ -978,14 +979,49 @@ void Editor::keyPressEvent(QKeyEvent* e)
       }
       else
       {
-        QMessageBox::critical(
-          this,
-          "Could not delete item",
-          "If deleting a vertex, it must not be in any edges or polygons "
-          "or lifts. "
-          "If deleting a feature, it must not be in any constraints.");
+        // Get the selected vertex index
+        int selected_vertex_idx = -1;
+        for (int i = 0;
+          i < static_cast<int>(building.levels[level_idx].vertices.size());
+          i++)
+        {
+          if (building.levels[level_idx].vertices[i].selected)
+          {
+            selected_vertex_idx = i;
+            break;
+          }
+        }
```

Comment:
Note to self but let's leave it to a followup, it is actually possible to do Ctrl-Click to select multiple vertices but the codebase only finds the first occurrence, we should consider whether to disallow selecting multiple vertices or make sure deletion can operate on an array of selected elements.

Reply:
I think the selection of multiple vertices is required to collinear the selected vertices, so disabling it might be tough.